File: python/lesson 6/level_order.py
# https://leetcode.com/problems/binary-tree-level-order-traversal/

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        if root == None: return result
        
        queue = deque()
        queue.append(root)
        
        while len(queue) > 0:
            size = len(queue)
            level = []
            for i in range(size):
                node = queue.popleft()
                level.append(node.val)
                if node.left != None:
                    queue.append(node.left)
                if node.right != None: 
                    queue.append(node.right)
            result.append(level)
        return result

File: ruby/lesson 6/level_order.rb
# https://leetcode.com/problems/binary-tree-level-order-traversal/

def level_order(root)
    result = []
    return result if root.nil?

    queue = Queue.new
    queue.push(root)

    while !queue.empty?
        size = queue.length
        level = []
        for i in 0..size-1
            node = queue.deq
            level.push(node.val)
            queue.push(node.left) unless node.left.nil?
            queue.push(node.right) unless node.right.nil?
        end
        result.push(level)
    end
    result
end
